XenAdmin/Wizards/PatchingWizard/PatchingWizard_SelectPatchPage.cs

```diff
@@ -173,10 +177,10 @@ public WizardMode WizardMode
             {
                 if (AutomatedUpdatesRadioButton.Visible && AutomatedUpdatesRadioButton.Checked)
                     return WizardMode.AutomatedUpdates;
-                var updateAlert = downloadUpdateRadioButton.Checked && dataGridViewPatches.SelectedRows.Count > 0
-                    ? (XenServerPatchAlert) ((PatchGridViewRow) dataGridViewPatches.SelectedRows[0]).UpdateAlert
+                var updateAlert = downloadUpdateRadioButton.Checked
+                    ? SelectedUpdateAlert
                     : selectFromDiskRadioButton.Checked
-                        ? GetAlertFromFileName(isValidFile(unzippedUpdateFilePath) ? unzippedUpdateFilePath.ToLowerInvariant() : fileNameTextBox.Text.ToLowerInvariant())

```

Comment:
Are we missing the isValidFile check with the new code?
In , I didn't find any place in GetAlertFromFile() where we check file name with "isValidFile(unzippedUpdateFilePath) ? unzippedUpdateFilePath.ToLowerInvariant() : fileNameTextBox.Text.ToLowerInvariant()"

Reply:
Sorry, I saw it. The check is in L221.
Forget it.